src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -180,9 +189,16 @@ public async Task StopReceive(CancellationToken cancellationToken = default)
                 {
                     connectionShutdownCompleted.SetResult(true);
                 }
+                connection.Dispose();
```

Comment:
Shouldn't the order be something like

```csharp
            messagePumpCancellationTokenSource?.Cancel();
            messagePumpCancellationTokenSource?.Dispose();
            messageProcessingCancellationTokenSource?.Cancel();
            messageProcessingCancellationTokenSource?.Dispose();

            connection?.Dispose();
           circuitBreaker?.Dispose();
```
and also have the connection dispose outside the using block together with the other disposes?

Reply:
hm I thought it might be related to triggering the shutdown event that is connected to the task completion source but I think it's not necessary.